drivers: ethernet: eth_nxp_enet_qos: remove unused variables

DerekSnell opened the pull request:
Read MAC registers to clear flags without using variables to avoid compiler warnings.

Resolves #91130 , which is blocking PRs with CI build failures

decsny reviewed (changes requested):
@maass-hamburg is correct, this will break the driver completely and cause infinite interrupt

decsny commented:
> > according to the comment above it reading them is needed to clear them, maybe use something like `(void)sys_read32(val);` instead?
> 
> Doh! Now I feel stupid. Thank you for pointing this out.

to be fair, I really don't like clear on read registers. although I think if I remember correctly, this hardware has a setting which you can change to so that it isn't read to clear. maybe look into that for this PR so we can make more sensible code, if I'm remembering a thing that really exists

theadib commented:
Hi @DerekSnell  some bits in the registers are reset while reading.

The previous code had a line:
`
-       volatile uint32_t mac_interrupts = base->MAC_INTERRUPT_STATUS;
-	volatile uint32_t mac_rx_tx_status = base->MAC_RX_TX_STATUS;
-	volatile uint32_t dma_interrupts = base->DMA_INTERRUPT_STATUS;
-	volatile uint32_t dma_ch0_interrupts = base->DMA_CH[0].DMA_CHX_STAT;
+	uint32_t mac_interrupts = base->MAC_INTERRUPT_STATUS;
+	uint32_t mac_rx_tx_status = base->MAC_RX_TX_STATUS;
+	uint32_t dma_interrupts = base->DMA_INTERRUPT_STATUS;
+	uint32_t dma_ch0_interrupts = base->DMA_CH[0].DMA_CHX_STAT;

-	mac_interrupts; mac_rx_tx_status;
`
That probably tricked the unused variable warning

maybe a simple touch is ok. since the base type should be volatile.
`
base->MAC_INTERRUPT_STATUS;
base->MAC_RX_TX_STATUS;
`
to be checked in the generated assembly lines. if the compiler keeps the instructions.
but putting back this is safer to circumvent the unused warning:
`
	mac_interrupts; mac_rx_tx_status;
`
Sorry my bad.

Commit:
drivers: ethernet: eth_nxp_enet_qos_mac: remove unused variables

Read MAC registers to clear flags without using variables to avoid
compiler warnings.

Signed-off-by: Derek Snell <derek.snell@nxp.com>
